### PR TITLE
docs: add Landlord description & Constraints

### PR DESCRIPTION
**Pull request by Akikaze1119**
## Overview
I add a description and constraints of 'Owner'

## Changes
▼screenshot
Screenshot 2024-02-03 at 9 31 53 PM

## Assignee Checklist:

- [x] The base branch is correct (no accidental merges)

## Reviewer Checklist

- Code readability and simplicity
- Follows best practices and coding standards
- Understandable and maintainable code

### CODE REVIEW

File: docs/docs/modeling.md [hunk 1]
```diff
@@ -103,7 +103,24 @@ classDiagram
 # Terms and Constraints
 
 ## Owner - オーナー
-
+### Owner
+The Owner is the Owner of ShareHouse
```
Comment (akiomatic) [inline docs/docs/modeling.md:107]: You might want to mention the `Owner` contains `ShareHouse`.

File: docs/docs/modeling.md [hunk 2]
```diff
@@ -103,7 +103,24 @@ classDiagram
...
+
+Constraints:
+Password must be
+- at least 8 characters
+- at least one capital letter
+- at least lowercase letter
+- at least one special character
+- at least one number
+### ShareHouse IDs
+The Owner is able to contain ShareHouse. The Owner class has ShareHouse ID list.
```
Comment (akiomatic) [inline docs/docs/modeling.md:123]: Since there is no relationship between `ShareHouseId` and `Owner`, I guess this part is not necessary.

**Review by akiomatic — CHANGES_REQUESTED** [review]
Thank you for adding the description and constraints for `Owner`.
I kindly ask you to take a look at my comments.

File: docs/docs/modeling.md (hunk 2 above)
Comment (Akikaze1119) [reply docs/docs/modeling.md]: I have removed the part that may duplicate the aspect mentioned in the initial description, "The Owner contains ShareHouse."

File: docs/docs/modeling.md [hunk 3]
```diff
@@ -103,7 +103,24 @@ classDiagram
...
+### Password
+The Owner has a password.
+
+Constraints:
+Password must be
+- at least 8 characters
+- at least one capital letter
+- at least lowercase letter
+- at least one special character
+- at least one number
```
[the remark this answers is not in the document]
Comment (Akikaze1119) [reply docs/docs/modeling.md]: I have standardized the representation to numeric format.

File: docs/docs/modeling.md [hunk 4]
```diff
@@ -103,7 +103,24 @@ classDiagram
 # Terms and Constraints
 
 ## Owner - オーナー
-
+### Owner
+The Owner is the Owner of ShareHouse
+### Owner ID (Identifier)
+The Owner ID is the UUID.
+### Email Address (Value Object)
+The Owner has an email address.
```
[the remark this answers is not in the document]
Comment (Akikaze1119) [reply docs/docs/modeling.md]: I have added the email address constraints, but I am concerned about potential gaps. I would like to discuss it in the meeting.

File: docs/docs/modeling.md [hunk 5]
```diff
@@ -104,6 +104,35 @@ classDiagram
 
 ## Owner - オーナー
 
+### Owner
+
+The Owner contains ShareHouse.
+
+### Owner ID (Identifier)
+
+The Owner ID is a UUID.
```
Comment (KaiKoide) [inline docs/docs/modeling.md:113]: Is it supposed to be a UUID?

Comment (yuki-92c) [reply docs/docs/modeling.md:113]: I guess "unique identifier (UID) " is appropriate in this context.

**Review by yuki-92c — COMMENTED** [review]
> ## Owner - オーナー

Before merging this branch, don't forget to change the word "Owner" to "Landlord".

File: docs/docs/modeling.md (hunk 5 above)
Comment (KaiKoide) [reply docs/docs/modeling.md:113]: As we discussed, we decided to make use of UUID as a ID.

File: docs/docs/modeling.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (Akikaze1119) [reply docs/docs/modeling.md]: @akiomatic 
I made revisions based on the decision made during the meeting to add constraints regarding uniqueness and the correct format for email addresses. Could you please review and confirm?

File: docs/docs/modeling.md [hunk 6]
```diff
@@ -103,7 +103,24 @@ classDiagram
...
+### Email Address (Value Object)
+The Owner has an email address.
+### Password
+The Owner has a password.
+
+Constraints:
+Password must be
+- at least 8 characters
+- at least one capital letter
+- at least lowercase letter
```
[the remark this answers is not in the document]
Comment (Akikaze1119) [reply docs/docs/modeling.md]: Thank you for your review. I revised that part.

File: docs/docs/modeling.md (hunk 1 above)
Comment (Akikaze1119) [reply docs/docs/modeling.md]: Thank you for your review. I revised that part.

File: docs/docs/modeling.md [hunk 7]
```diff
@@ -102,7 +102,38 @@ classDiagram
...
+
+The Landlord ID is a UUID.
+
+### Email Address (Value Object)
+
+The Landlord has an email address.
+
+Constraints:
+
+- unique. (Landlords cannot create multiple accounts with the same email address).
```
Comment (akiomatic) [inline docs/docs/modeling.md:121]: Could you replace the `unique` with `uniqueness`?

**Review by akiomatic — CHANGES_REQUESTED** [review]
Thank you for your modifications.
I kindly ask you to read my new comments.

File: docs/docs/modeling.md [hunk 8]
```diff
@@ -102,7 +102,38 @@ classDiagram
...
-## Owner - オーナー
+## Landlord - 大家
+
+### Landlord
+
+The Landlord contains ShareHouse.
+
+### Landlord ID (Identifier)
+
+The Landlord ID is a UUID.
```
[the remark this answers is not in the document]
Comment (Akikaze1119) [reply docs/docs/modeling.md:113]: @akiomatic 
Thank you for the review. Upon verification, it seems that using the indefinite article 'a' with 'UUID' is correct.

Reference:
https://developer.apple.com/documentation/foundation/uuid
https://www.techtarget.com/searchapparchitecture/definition/UUID-Universal-Unique-Identifier

**Review by akiomatic — APPROVED** [review]
LGTM!
I apologize for the latest review!

**Review by HrRn — APPROVED** [review]
Thank you!

**Review by yuki-92c — APPROVED** [review]
Thanks. Looks good to me:)

**Review by KaiKoide — APPROVED** [review]

**Review by akiomatic — APPROVED** [review]
LGTM!

**Pull request merged**